hera_cal/abscal_funcs.py

```diff
@@ -1113,6 +1112,66 @@ def interp2d_vis(model, model_lsts, model_freqs, data_lsts, data_freqs, flags=No
     return DataContainer(new_model), DataContainer(new_flags)
 
 
+def rephase_vis(model, model_lsts, data_lsts, bls, freqs, flags=None, max_dlst=0.005, latitude=-30.72152):
+    """
+    Rephase model visibility data onto LST grid of data_lsts.
+
+    Parameters:
+    -----------
+    model : type=DataContainer, holds complex visibility for model
+        keys are antenna-pair + pol tuples, values are 2d complex visibility
+        with shape (Ntimes, Nfreqs)
+
+    model_lsts : 1D array of the LST grid in model [radians], dtype=float, shape=(Ntimes,)
+
+    data_lsts : 1D array of the LST grid in data [radians], dtype=float, shape=(Ntimes,)
+
+    bls : type=dictionary, ant-pair keys that holds baseline position vector in ENU frame in meters
+
+    freqs : type=float ndarray, holds frequency channels of model in Hz.
+
+    flags : type=DataContainer, holds model flags
+
+    max_dlst : type=bool, maximum dlst [radians] to allow for rephasing, otherwise flag data.
+
+    latitude : type=float, latitude of array in degrees North
+
+    Return: (new_model, new_flags)
+    -------
+    new_model : DataContainer with rephased model
+    new_flags : DataContainer with new flags
+    """
+    # unravel LST array if necessary
+    if data_lsts.max() < data_lsts.min():
+        data_lsts[data_lsts]
+    # get nearest neighbor model points
+    lst_nn = np.array(map(lambda x: np.argmin(np.abs(model_lsts-x)), data_lsts))
+
+    # get dlst array
+    dlst = data_lsts - model_lsts[lst_nn]
+
+    # flag dlst above threshold
+    flag_lst = np.zeros_like(dlst, np.bool)
+    flag_lst[np.abs(dlst) > max_dlst] = True
+
+    # make new_model and new_flags
+    new_model = odict()
+    new_flags = odict()
+    for k in model.keys():
+        m = model[k][lst_nn, :]
+        new_model[k] = m
+        if flags is None:
+            new_flags[k] = np.zeros_like(m, np.bool)
+        else:
+            new_flags[k] = flags[k][lst_nn, :]
+        new_flags[k][flag_lst] = True
```

Comment:
Should this be `new_flags[k][flag_lst,:] = True`. It might work as is, but I find being very explicit about dimensions helps both code readability and debugging.